curation-migration-datasync/eventConsumer.ts

```diff
@@ -0,0 +1,123 @@
+import { DataService } from './database/dataService';
+import { getTopicForReaditLaTmpDatabase } from './helpers/topicMapper';
+import { getParserMetadata } from './externalCaller/parser';
+import { AddScheduledItemPayload, TileSource } from './types';
+import {
+  hydrateCuratedFeedItem,
+  hydrateCuratedFeedProspectItem,
+  hydrateCuratedFeedQueuedItem,
+} from './helpers/hydrator';
+import { CuratedItemRecord, ScheduledSurfaceGuid } from './dynamodb/types';
+import { insertCuratedItem } from './dynamodb/curatedItemIdMapper';
+import { dbClient } from './dynamodb/dynamoDbClient';
+import { Knex } from 'knex';
+
+/**
+ * function to generate epoc timestamp
+ * @param scheduledDate
+ */
+export function convertDateToTimestamp(scheduledDate: string) {
+  return Math.round(new Date(scheduledDate).getTime() / 1000);
+}
+
+/**
+ * fetches `cohara` from 'ad|Mozilla-LDAP|cohara'
+ * @param ssoName
+ */
+export function getCuratorNameFromSso(ssoName: string) {
+  const prefix = 'ad|Mozilla-LDAP|';
+  if (!ssoName.startsWith(prefix)) {
+    throw new Error(
+      'unexpected sso format, createdBy are expected to startWith `ad|Mozilla-LDAP|`'
+    );
+  }
+  return ssoName.substring(prefix.length);
+}
+
+/**
+ * converts the event body of `add-scheduled-item` event to database models,
+ * and inserts them to the database within one transaction. If successful,
+ * adds the curatedRecId and externalIds from the event to the dynamoDb
+ * @param eventBody
+ */
+export async function addScheduledItem(
```

Comment:
Is there a reason you separated this from the dataservice?

Reply:
ya addScheduledItem calls multiple functions, like parser, dynamo and rds calls - whereas i am seeing `dataService` as functions that access rds alone.

Reply:
hmm, to me I think it would be a lot easier if the query part was all maintained in the same service. This caller function could be responsible only for hydrating the data which should be inserted.

this would also simplify your transaction logic and I think make it easier to fix the test

Reply:
ya that make sense, will move the db query part to dataService